src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/GenericXpathCheck.java

```diff
@@ -0,0 +1,148 @@
+import net.sf.saxon.Configuration;
+import net.sf.saxon.om.Item;
+import net.sf.saxon.sxpath.XPathDynamicContext;
+import net.sf.saxon.sxpath.XPathEvaluator;
+import net.sf.saxon.sxpath.XPathExpression;
+import net.sf.saxon.trans.XPathException;
+
+/**
+ * <p>
+ * Evaluates XPath query and verifies absence of the corresponding nodes. All nodes
+ * retrieved by XPath evaluator will be logged as violations. Current check allows
+ * user to implement custom checks using XPath.
+ * </p>
+ * <ul>
+ * <li>
+ * Property {@code query} - XPath query. Type is {@code java.lang.String}. Default value
+ * is {@code ""}.
+ * </li>
+ * </ul>
+ * <p>
+ * To configure the check:
+ * </p>
+ * <pre>
+ * &lt;module name="GenericXpath"/&gt;
+ * </pre>
+ * <p>
+ * To fail with violation if method name is 'test' or 'foo'
+ * </p>
+ * <pre>
+ * &lt;module name="GenericXpath"&gt; &lt;property name="query"
+ * value="//METHOD_DEF[./IDENT[@text='test' or @text='foo']]"/&gt; &lt;/module&gt;
+ * </pre>
+ * <p>
+ * Example:
+ * </p>
+ * <pre>
+ * public class Test {
+ *  public void func1() throws RuntimeException {} // violation
+ *  public void func2() throws Exception {} // ok
+ *  public void func3() throws Error {} // violation
+ *  public void func4() throws Throwable {} // violation
+ *  public void func5() throws NullPointerException {} // ok
+ * }
+ * </pre>
+ * <p>
+ * Parent is {@code com.puppycrawl.tools.checkstyle.TreeWalker}
+ * </p>
+ * <p>
+ * Violation Message Keys:
+ * </p>
+ * <ul>
+ * <li>
+ * {@code xpath.finding}
+ * </li>
+ * </ul>
+ *
+ * @since 8.37
+ */
+@StatelessCheck
+public class GenericXpathCheck extends AbstractCheck {
+
+    /**
+     * A key is pointing to the warning message text provided by user.
+     */
+    public static final String MSG_KEY = "xpath.finding";
+
+    /** XPath query. */
+    private String query = "";
+
+    @Override
+    public void beginTree(DetailAST rootAST) {
+        if (!query.isEmpty()) {
+            evaluatexPathQuery(rootAST);
+        }
+    }
+
+    private void evaluatexPathQuery(DetailAST rootAST) {
+        final XPathEvaluator xpathEvaluator = new XPathEvaluator(Configuration.newConfiguration());
+        try {
+            final RootNode rootNode = new RootNode(rootAST);
+            final XPathExpression xpathExpression = xpathEvaluator.createExpression(query);
+            final XPathDynamicContext xpathDynamicContext =
+                    xpathExpression.createDynamicContext(rootNode);
+            final List<Item> matchingItems = xpathExpression.evaluate(xpathDynamicContext);
+
+            matchingItems.forEach(node -> {
+                final AbstractNode xpathNode = (AbstractNode) node;
+                log(xpathNode.getUnderlyingNode(), MSG_KEY);
+            });
+        }
+        catch (XPathException ex) {
+            throw new IllegalArgumentException(ex);
+        }
+    }
+
+    @Override
+    public int[] getDefaultTokens() {
+        return CommonUtil.EMPTY_INT_ARRAY;
```

Comment:
To make Pitest happy, call `getRequiredTokens` here. Like in this Check:

https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/NoCodeInFileCheck.java#L93

Reply:
done